### PR TITLE
zareen js exercises week4

### PR DESCRIPTION
**Pull request by khan-zareen**

### CODE REVIEW

File: week-4/E-arrays-of-objects/exercise-1.js [hunk 1]
```diff
@@ -5,42 +5,59 @@
...
+  return namesOf.name;
+});
 
-var persons = // Complete here
-
-var personNames = // Complete here
-
-var personsYoungerThan28YearsOld = // Complete here
-
+var personsYoungerThan28YearsOld = persons.filter(ages => {
```
Comment (nbogie) [inline week-4/E-arrays-of-objects/exercise-1.js]: `ages` is a misleading name for this parameter.  The parameter to the callback function given to persons.filter() should indicate its type.  The function doesn't get called with a list of ages.  What does it get called with?  What would be a better name?

File: week-4/E-arrays-of-objects/exercise-2.js [hunk 2]
```diff
@@ -7,50 +7,72 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
...
+var destinationNameReachableByFerry = travelDestinations
+  .filter(reach => reach.transportations.includes("ferry"))
+  .map(destination => destination.destinationName);
 
-var destinationNamesWithin500Kms = // Complete here
-
-var destinationNameReachableByFerry = // Complete here
-
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
+travelDestinations.forEach;
```
Comment (nbogie) [inline week-4/E-arrays-of-objects/exercise-2.js]: what does this line (53) do?

File: week-4/E-arrays-of-objects/exercise-2.js [hunk 3]
```diff
@@ -7,50 +7,72 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
...
+  destination4
+];
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(destination => destination.distanceKms < 500)
+  .map(destination => destination.destinationName);
```
Comment (nbogie) [inline week-4/E-arrays-of-objects/exercise-2.js]: good

File: week-4/E-arrays-of-objects/exercise-2.js [hunk 4]
```diff
@@ -7,50 +7,72 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
...
 
+var destinationNameReachableByFerry = travelDestinations
+  .filter(reach => reach.transportations.includes("ferry"))
+  .map(destination => destination.destinationName);
```
Comment (nbogie) [inline week-4/E-arrays-of-objects/exercise-2.js]: reach isn't a good name for the parameter to the callback function given to filter.  filter will take each destination in turn and check it against the callback function.  Can you think of a better parameter name?

File: week-4/E-arrays-of-objects/exercise-2.js [hunk 5]
```diff
@@ -7,50 +7,72 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
...
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
+travelDestinations.forEach;
 
+var destinationNamesMoreThan300KmsAwayByTrain = travelDestinations
+  .filter(
+    destination =>
+      destination.distanceKms > 300 &&
+      destination.transportations.includes("train")
+  )
+  .map(destinationName => destinationName.destinationName);
```
Comment (nbogie) [inline week-4/E-arrays-of-objects/exercise-2.js]: `destinationName.destinationName`... doesn't make sense.  The name of a destination (perhaps 'Paris') is a string and strings don't have a `name` property, in the same way that *your* name doesn't have a name.
Tip: if you find yourself writing phrases which sound nonsensical, stop and check that your variables are well named and that your logic is good.

But, it works, well done!  Now try to make it read better.

File: week-4/E-arrays-of-objects/exercise-3.js [hunk 6]
```diff
@@ -53,33 +53,57 @@ var restaurants = [restaurant1, restaurant2, restaurant3];
...
+        restaurant =>
+          restaurant.totalSeats - restaurant.numberOfCustomers > numberOfPeople
+      )
+      .map(displayName);
+  },
+  findRestaurantServingDish: function(dishName) {
+    return restaurants
+      .filter(restaurant => restaurant["menu"].includes(dishName))
+      .map(displayName);
+  },
```
Comment (nbogie) [inline week-4/E-arrays-of-objects/exercise-3.js]: good.  I'd probably have said `restaurant.menu` here, not `restaurant["menu"]`

File: week-4/E-arrays-of-objects/exercise-3.js [hunk 7]
```diff
@@ -53,33 +53,57 @@ var restaurants = [restaurant1, restaurant2, restaurant3];
...
+    restaurantsServingSalad
+);
+
+var numberOfRestaurantsInCityCentre = restaurantFinderApplication.countNumberOfRestaurantsInArea(
+  "center"
+);
+console.log(
+  "Number of restaurants in city centre: Expected result: 2, actual result: " +
+    numberOfRestaurantsInCityCentre
+);
```
Comment (nbogie) [inline week-4/E-arrays-of-objects/exercise-3.js]: great stuff!

File: week-4/E-arrays-of-objects/exercise-1.js [hunk 8]
```diff
@@ -5,42 +5,59 @@
...
 };
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
+var persons = [person1, person2, person3];
+console.log(persons);
 
+var personNames = persons.map(namesOf => {
```
[the remark this answers is not in the document]
Comment (khan-zareen) [reply week-4/E-arrays-of-objects/exercise-1.js]: yes you right, i modified it

File: week-4/E-arrays-of-objects/exercise-2.js (hunk 2 above)
Comment (khan-zareen) [reply week-4/E-arrays-of-objects/exercise-2.js]: It was mistake i removed it

File: week-4/E-arrays-of-objects/exercise-2.js (hunk 4 above)
Comment (khan-zareen) [reply week-4/E-arrays-of-objects/exercise-2.js]: destination

**Pull request closed without merging**